programs/kyupad-ido/src/instructions/admin.rs

```diff
@@ -0,0 +1,81 @@
+use anchor_lang::prelude::*;
+
+use crate::Master;
+
+pub fn add_admin(_ctx: Context<AddAdmin>, _address: Pubkey) -> Result<()> {
+    Ok(())
+}
+
+pub fn delete_admin(ctx: Context<DeleteAdmin>, _address: Pubkey) -> Result<()> {
+    let signer = &mut ctx.accounts.signer;
+    let admin_pda = &ctx.accounts.admin_pda;
+
+    let dest_starting_lamports = signer.lamports();
+    **signer.lamports.borrow_mut() = dest_starting_lamports
+        .checked_add(admin_pda.lamports())
+        .unwrap();
+
+    **admin_pda.lamports.borrow_mut() = 0;
+    let mut source_data = admin_pda.data.borrow_mut();
+    source_data.fill(0);
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+#[instruction(_address: Pubkey)]
+pub struct AddAdmin<'info> {
+    #[account(
+        mut,
+        constraint = signer.key() == master_pda.master_key
```

Comment:
What will going on? if:
- I have a key X
- I init a master_pda with same data structure
- Assign my self wallet id as master_key

Reply:
- Chỉ có duy nhất 1 người được tạo master_pda, đó chính là tk deployer
- Cái master_pda chỉ được tạo đúng duy nhất 1 lần
- Nếu giả sử key của mình là master_key thì mình có thể thề làm add_admin, xoá admin, chuyển quyền master sang người khác
- Nếu tk master mà muốn tạo project, hay sửa config của project thì phải tạo 1 tài khoản admin cho chính nó

Reply:
> Cái master_pda chỉ được tạo đúng duy nhất 1 lần

Yup, but what hapend when some one (not our deployer) make a same PDA with the data same as Master struct and pass it there?

Reply:
Cannot sir